src/interfaces/Ui.ts

```diff
@@ -0,0 +1,14 @@
+export interface Props {
+  children: React.ReactNode;
+}
+
+export interface ButtonProps extends Props {
+  type?: "button" | "submit" | "reset" | undefined;
```

Comment:
do you need to type "undefined" explicitly if you already marked the property as optional?

Reply:
Changed